scrapy/utils/misc.py

```diff
@@ -116,4 +116,28 @@ def md5sum(file):
 def rel_has_nofollow(rel):
     """Return True if link rel attribute has nofollow type"""
     return True if rel is not None and 'nofollow' in rel.split() else False
-    
+
+def construct_from_settings_or_crawler(objcls, settings, crawler, *args,
```

Comment:
woops didn't even noticed it was a util function. I think the point is we should deprecate the few instances when this happens, so there is no point on having such a function. this makes sense?